src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -643,22 +644,21 @@ private static Array BindArray(Array source, IConfiguration config, BinderOption
 
         private static List<PropertyInfo> GetAllProperties(
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
-            Type type)
+            Type? type)
         {
             var allProperties = new List<PropertyInfo>();
 
-            do
+            while (type != null && type != typeof(object))
```

Comment:
Is this change necessary? It seems to be changing behavior.

Maybe the `Type type` parameter shouldn't be nullable?